### PR TITLE
Fix escaping, add https pattern matches

### PR DESCRIPTION
**Pull request by piatra**
So bizarre: the escaping issue is not showing up in the diff :/ if this still doesn't work I will manually fix it when deploying to RS.
Also added match for `https` and not enrolling if users specified that in their preferences.

Details: https://docs.google.com/document/d/1-6yFzp3iQKbaeQl624O3a8GN3DGz7eg4rvQI18BS1nE/edit#heading=h.e871marbqms9

### CODE REVIEW

File: messaging-experiments.json [hunk 1]
```diff
@@ -265,8 +265,8 @@
   {
     "id": "PROTECTION-RETENTION-EXPERIMENT-RELEASE",
     "enabled": true,
-    "filter_expression": "env.channel == 'nightly' || env.channel == 'default'",
-    "targeting": "localeLanguageCode == 'en' && [userId, \"PROTECTION-RETENTION-EXPERIMENT-RELEASE\"]|stableSample(0.1)",
+    "filter_expression": "(env.channel == 'release' || env.channel == 'default') && env.version >= '77.'",
```
Comment (ncloudioj) [inline messaging-experiments.json:268]: Is the dot intended here in `77.`? To exclude `77`?

Just to confirm, do we have the required strings for those messages in 77?

File: messaging-experiments.yaml [hunk 2]
```diff
@@ -200,7 +200,7 @@
               label:
                 string_id: cfr-doorhanger-extension-sumo-link
               sumo_path: extensionrecommendations
-            text: You’re using the only major browser backed by a non-profit. See how Firefox puts your privacy first.
+            text: 'You’re using the only major browser backed by a non-profit. See how Firefox puts your privacy first.'
```
Comment (ncloudioj) [inline messaging-experiments.yaml:203]: How about using double quotes?

File: messaging-experiments.json (hunk 1 above)
Comment (piatra) [reply messaging-experiments.json:268]: > https://github.com/mozilla/messaging-system-inflight-assets/blob/daf93389d7b4065871f70aa0591614634fca61af/messaging-experiments.yaml#L3

This is [Remote Settings syntax](https://searchfox.org/mozilla-central/rev/91d82d7cbf05a71954dfa49d0e43824c7c973e62/services/settings/remote-settings.js#70-73)
https://github.com/mozilla/messaging-system-inflight-assets/blob/daf93389d7b4065871f70aa0591614634fca61af/messaging-experiments.yaml#L3

> Just to confirm, do we have the required strings for those messages in 77?

Messages are `en` only and don't reference Fluent strings.

File: messaging-experiments.yaml (hunk 2 above)
Comment (piatra) [reply messaging-experiments.yaml:203]: Quotes don't seem to help, I can fix this when publishing to RS. I dropped the quotes entirely.

**Pull request merged**